lib/cluster/election.c

```diff
@@ -638,7 +639,10 @@ election_count_vote(election_t *e, xmlNode *message, bool can_win)
                      e->name, election_wins, STORM_INTERVAL);
             election_wins = 0;
             expires = tm_now + STORM_INTERVAL;
-            crm_write_blackbox(0, NULL);
+            if (wrote_blackbox == FALSE) {
+                wrote_blackbox = TRUE; // Don't add a flood of blackboxes
+                crm_write_blackbox(0, NULL);
+            }
```

Comment:
I went back and forth on this a bit. Two other possibilities are once-per-time-period (e.g. track last_blackbox timestamp and write if more than 1 hour ago), or once-per-n-floods (e.g. track int flood_count instead of wrote_blackbox and write when flood_count++ % 10 == 0).

I decided against mainly because I don't see a whole lot of value in getting a black box here at all. The main use I think would be during development/testing to debug something that went wrong, so one should be sufficient. I wouldn't expect that a production environment would report multiple floods of different causes.

I think the flood detection is worthwhile for generating a log, and maybe it would be nice to adjust behavior, though I'm not sure what we could do. But the idea of tracing through even a single black box from every node to figure out what sequence caused an election storm is pretty horrifying. I left it in case I'm wrong :) but I could be convinced to either remove it entirely or go with one of the other methods.

Reply:
Would expect a comment like "once per life-time occurrence is more that enough" then,
to make the decision explicit.